gm4_dripleaf_filters/data/gm4_dripleaf_filters/functions/check_hopper_filter.mcfunction

```diff
@@ -0,0 +1,38 @@
+execute unless score $filters_match gm4_dripleaf_filters matches 1 unless score $hopper_checked gm4_dripleaf_filters matches 1 store success score $items_are_different gm4_dripleaf_filters run data modify storage gm4_dripleaf_filters:cache hopperCompare[0].id set from storage gm4_dripleaf_filters:cache itemCompare
+execute unless score $filters_match gm4_dripleaf_filters matches 1 unless score $hopper_checked gm4_dripleaf_filters matches 1 if score $items_are_different gm4_dripleaf_filters matches 0 run scoreboard players set $filters_match gm4_dripleaf_filters 1
+execute unless score $filters_match gm4_dripleaf_filters matches 1 unless score $hopper_checked gm4_dripleaf_filters matches 1 run data remove storage gm4_dripleaf_filters:cache hopperCompare[0]
+# slot 3
+execute unless score $filters_match gm4_dripleaf_filters matches 1 if data storage gm4_dripleaf_filters:cache {hopperCompare:[]} run scoreboard players set $hopper_checked gm4_dripleaf_filters 1
+execute unless score $filters_match gm4_dripleaf_filters matches 1 unless score $hopper_checked gm4_dripleaf_filters matches 1 store success score $items_are_different gm4_dripleaf_filters run data modify storage gm4_dripleaf_filters:cache hopperCompare[0].id set from storage gm4_dripleaf_filters:cache itemCompare
+execute unless score $filters_match gm4_dripleaf_filters matches 1 unless score $hopper_checked gm4_dripleaf_filters matches 1 if score $items_are_different gm4_dripleaf_filters matches 0 run scoreboard players set $filters_match gm4_dripleaf_filters 1
+execute unless score $filters_match gm4_dripleaf_filters matches 1 unless score $hopper_checked gm4_dripleaf_filters matches 1 run data remove storage gm4_dripleaf_filters:cache hopperCompare[0]
+# slot 4
+execute unless score $filters_match gm4_dripleaf_filters matches 1 if data storage gm4_dripleaf_filters:cache {hopperCompare:[]} run scoreboard players set $hopper_checked gm4_dripleaf_filters 1
+execute unless score $filters_match gm4_dripleaf_filters matches 1 unless score $hopper_checked gm4_dripleaf_filters matches 1 store success score $items_are_different gm4_dripleaf_filters run data modify storage gm4_dripleaf_filters:cache hopperCompare[0].id set from storage gm4_dripleaf_filters:cache itemCompare
+execute unless score $filters_match gm4_dripleaf_filters matches 1 unless score $hopper_checked gm4_dripleaf_filters matches 1 if score $items_are_different gm4_dripleaf_filters matches 0 run scoreboard players set $filters_match gm4_dripleaf_filters 1
+execute unless score $filters_match gm4_dripleaf_filters matches 1 unless score $hopper_checked gm4_dripleaf_filters matches 1 run data remove storage gm4_dripleaf_filters:cache hopperCompare[0]
+
+execute if score $filters_match gm4_dripleaf_filters matches 1 run function gm4_dripleaf_filters:deposit_item

```

Comment:
This could be done recursively, saving on one command per slot and making this a lot shorter. For this you have to create one function `match_item`:
```
execute store success score $items_match gm4_dripleaf_filters run data modify storage gm4_dripleaf_filters:cache hopperCompare[0].id set from storage gm4_dripleaf_filters:cache itemCompare
execute if score $items_match gm4_dripleaf_filters matches 0 run data remove storage gm4_dripleaf_filters:cache hopperCompare[0]
execute if score $items_match gm4_dripleaf_filters matches 0 if data storage gm4_dripleaf_filters:cache hopperCompare[0] run function gm4_dripleaf_filters:match_item
```

That is called from `check_hopper_filter`:
```suggestion
# loop through the slots to see if any IDs match
function gm4_dripleaf_filters:match_item
execute if score $items_match gm4_dripleaf_filters matches 1 run function gm4_dripleaf_filters:deposit_item
```

Note that I have not tested this in-game.